prog15.rb

```diff
@@ -0,0 +1,24 @@
+# Define a method to find all prime numbers upto n using 'step' function. Don't use Prime class
+# [input]
+# 20
+# [output]
+# [2, 3, 5, 7, 11, 13, 17, 19] 
+
+def prime(n)
+  (2..n).step(1).each do |value|
+    is_prime=true
+    (2..Math.sqrt(value)).each do |e|
+      if value%e==0
+        is_prime=false
+        break
+      end 
+    end
+    if is_prime
+      yield value
+    end
+  end
+end
+
+num=gets.to_i
+prime(num){|e| print e , ","}
```

Comment:
Instead of printing the prime numbers, push them into an array and print that array at the end. This is how the output is expected (in an array).